src/connectors/numeric-menu/__tests__/connectNumericMenu-test.ts

```diff
@@ -1017,4 +1017,218 @@ See documentation: https://www.algolia.com/doc/api-reference/widgets/numeric-men
       );
     });
   });
+
+  describe('getRenderState', () => {
+    it('returns the render state', () => {
+      const [widget, helper] = getInitializedWidget();
+
+      const renderState1 = widget.getRenderState(
+        {},
+        createInitOptions({ state: helper.state, helper })
+      );
+
+      expect(renderState1.numericMenu).toEqual({
+        numerics: {
+          createURL: expect.any(Function),
+          hasNoResults: true,
+          items: [
+            {
+              isRefined: false,
+              label: 'below 10',
+              value: '%7B%22end%22:10%7D',
+            },
+            {
+              isRefined: false,
+              label: '10 - 20',
+              value: '%7B%22start%22:10,%22end%22:20%7D',
+            },
+            {
+              isRefined: false,
+              label: 'more than 20',
+              value: '%7B%22start%22:20%7D',
+            },
+          ],
+          refine: expect.any(Function),
+          widgetParams: {
+            attribute: 'numerics',
+            items: [
+              {
+                end: 10,
+                label: 'below 10',
+              },
+              {
+                end: 20,
+                label: '10 - 20',
+                start: 10,
+              },
+              {
+                label: 'more than 20',
+                start: 20,
+              },
+            ],
+          },
+        },
+      });
+
+      const results = new SearchResults(helper.state, [
+        createSingleSearchResponse(),
+      ]);
+
+      const renderState2 = widget.getRenderState(
+        {},
+        createRenderOptions({
+          helper,
+          state: helper.state,
+          results,
+        })
+      );
+
+      expect(renderState2.numericMenu).toEqual({
+        numerics: {
+          createURL: expect.any(Function),
+          refine: expect.any(Function),
```

Comment:
Maybe use:

```suggestion
          refine: renderState1.numericMenu.numerics.refine,
```

Just to be sure we get the same function as in the first render ?